Sources/Profile-Modify.php

```diff
@@ -3161,7 +3161,7 @@ function profileSaveAvatarData(&$value)
 		$url = parse_url($_POST['userpicpersonal']);
 		$contents = fetch_web_data($url['scheme'] . '://' . $url['host'] . (empty($url['port']) ? '' : ':' . $url['port']) . str_replace(' ', '%20', trim($url['path'])));
 
-		$new_filename = $uploadDir . '/' . getAttachmentFilename('avatar_tmp_' . $memID, false, null, true);
+		$new_filename = $uploadDir . '/' . getAttachmentFilename('avatar_tmp_' . $memID, null, false, true);
```

Comment:
With this change you would be the `$attachment_id` parameter to `null` and the `$dir` parameter to `false`. Changing `$attachment_id` doesn't make any difference, since it will just be cast to an int. But the changing `$dir` from `null` to `false` will cause problems. `$dir` is used as a key to the `$modSettings['attachmentUploadDir']` array. A `null` key will return nothing, but a `false` key will be treated like a `0` key, i.e. the first key of the array. That's not what we want in `getAttachmentFilename()`.

The same applies to the other similar changes in this PR.